torch/csrc/autograd/engine.cpp

```diff
@@ -1117,13 +1117,38 @@ void Engine::add_thread_pool_task(const std::weak_ptr<GraphTask>& graph_task) {
 }
 
 void GraphTask::init_to_execute(Node& graph_root, const edge_list& outputs, bool accumulate_grad) {
+  // Populates exec_info so nodes that should be executed have `exec_info[node].needed_ = true`
+  // Only nodes that have a path to any edge in `outputs` should be executed.
+  // The code below populates exec_info using recursion, but the actual code does this
+  // iteratively. Refer to the numbering to see how the actual code corresponds.
+  //
+  // is_needed = {fn: True for fn in outputs}             # (0)
```

Comment:
nit: the naming here look like this would be a boolean.
Also is there a reason to use a dictionary instead of a unordered set here? Or is it just to mimick the exec_info_ that is actually a map?

Reply:
Yep, its just to mimic exec_info_.

Reply:
I guess `is_needed` does seem like a bool, but thats what it was before and I couldn't come up with a better name. The dict technically does contain boolean values, so when you read `is_needed[fn]`, I think it does make sense.